pkg/controller.v1beta1/trial/trial_controller_test.go

```diff
@@ -179,137 +175,146 @@ func TestReconcileBatchJob(t *testing.T) {
 		},
 	}
 
-	mockManagerClient.EXPECT().GetTrialObservationLog(gomock.Any()).Return(observationLogAvailable, nil).Times(1)
-	mockManagerClient.EXPECT().GetTrialObservationLog(gomock.Any()).Return(observationLogUnavailable, nil).MinTimes(1)
-	mockManagerClient.EXPECT().DeleteTrialObservationLog(gomock.Any()).Return(nil, nil).AnyTimes()
+	t.Run(`Trial run with "Failed" BatchJob.`, func(t *testing.T) {
+		g := gomega.NewGomegaWithT(t)
+		mockMC.msg = observationLogUnavailable
```

Comment:
```suggestion
		mockManagerClient.EXPECT().GetTrialObservationLog(gomock.Any()).Return(observationLogAvailable, nil).AnyTimes()
```

Instead of this original mock client, can we reuse the autogenerated mock clients like this?

Reply:
We need [WithOverridableExpectations](https://pkg.go.dev/go.uber.org/mock@v0.4.0/gomock#WithOverridableExpectations) here, but github.com/golang/mock/mock does not support this feature.
Or ["Each test should create a new Controller"](https://pkg.go.dev/github.com/golang/mock@v1.6.0/gomock#Controller).
In my opinion, gomock is not flexible enough, and it complicates a simple thing for this scenario.

Reply:
Finally, I found another way.

Reply:
@forsaken628 I opened the issue to replace gomock with go.uber.com/gomock here: https://github.com/kubeflow/katib/issues/2356

As a separate PR, could you work on the issue?